test/models/user_test.rb

```diff
@@ -718,11 +718,11 @@ def setup_user(operation, type = 'users', search = nil)
     not_existing_auth_source = 'new_external_source'
 
     context "internal or not existing AuthSource" do
-      test 'existing user' do
+      test 'existing user without auth source specified' do
         assert_difference('User.count', 0) do
-          login = users(:one).login
-          assert_equal User.find_or_create_external_user({:login => login}, nil),
-            User.find_by_login(login)
+          login = users(:external).login
+          user = User.find_or_create_external_user({:login => login}, nil)
+          assert user, users(:external)
```

Comment:
This should be:
```suggestion
          assert_equal user, users(:external)
```
Otherwise, it will fail if `user` is falsy and display `users(:external)` as the message

Reply:
there you go! :+1:  thanks for all the nits!